src/core/execute.c

```diff
@@ -436,7 +436,7 @@ int exec_spawn(Unit *unit,
         /* If LogLevelMax= is specified, then let's use the specified log level at the beginning of the
          * executor process. To achieve that the specified log level is passed as an argument, rather than
          * the one for the manager process. */
-        r = log_max_levels_to_string(context->log_level_max >= 0 ? context->log_level_max : log_get_max_level(), &max_log_levels);
+        r = log_max_levels_to_string(context->log_level_max, &max_log_levels);
```

Comment:
or hmm, do we use log_level_max = -1 as indicator for no clamping already here? hmm, maybe using -1 is ok then. man this is a bit confusing to me